packages/@headlessui-react/src/components/combobox/combobox.tsx

```diff
@@ -1736,7 +1753,17 @@ function OptionsFn<TTag extends ElementType = typeof DEFAULT_OPTIONS_TAG>(
       >
         {render({
           ourProps,
-          theirProps,
+          theirProps: {
+            ...theirProps,
+            children: (
+              <Frozen freeze={visible && data.comboboxState === ComboboxState.Closed}>
+                {typeof theirProps.children === 'function'
+                  ? // @ts-expect-error The `children` prop now is a callback function
+                    theirProps.children?.(slot)
+                  : theirProps.children}
+              </Frozen>
+            ),
+          },
```

Comment:
Had to do this here to make sure we only freeze the `children`, and not the `ComboboxOptions` itself (aka, the `return render()`)

Reply:
It feels a bit weird because it kinda feels like this should still be the responsibility of render (or a similar helper). But, having said that, this is fine for now.